diego_villanueva/week01/mta_lab2/js/main.js

```diff
@@ -0,0 +1,119 @@
+// All 3 subway lines intersect at Union Square, but there are no other intersection points.
+//  (For example, this means the 28th stop on the N line is different than the 28th street stop on
+// the 6 line, so you'll have to differentiate this when you name your stops in the arrays.)
+// Tell the user the number of stops AND the stops IN ORDER that they will pass through or change at.
+// Hints:
+//
+// Work out how you would do it on paper first! Then start to explain that process in Javascript.
+// Get the program to work for a single line before trying to tackle multiple lines.
+// Don't worry about prompting the user for input. Hard code some values to get it working.
+//  You can use prompt() later to make it more interactive.
+// Consider diagramming the lines by sketching out the subway lines and their stops and intersection.
+// The key to the lab is finding the index positions of each stop. (hint: indexOf())
+// Make sure the stops that are the same for different lines have different names (i.e. 23rd
+//   on the N and on the 6 need to be differentiated)
+
+
+var trip = [];
+var numberOfStops = 0;
+
+var lines = {
+  n: ['Times Square', '34th', '28th', '23rd', 'Union Square', '8th'],
+  l: ['8th', '6th', 'Union Square', '3rd', '1st'],
+  6: ['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place']
+};
+
+var collectData = function (){
+  line1 = prompt('Select a subway line to start, options are: n, l and 6','n');
+  station1 = prompt('Select one of the following stations: ' + lines[line1] + '.');
+  line2 = prompt('Select a subway line to start, options are: n, l and 6','n');
+  station2 = prompt('Select one of the following stations: ' + lines[line2] + '.');
+};
+
+// line1, station1, line2, station2
+
+var findPosition = function(line, station){  //it finds the position in the array of the stations / a number
+  return lines[line].indexOf(station);
+};
+
+
+
+var findDirection = function( location1, location2){  // it tells you whether you should go right or left to get to the other station, and how many stations
+  return location2 - location1      // if it is negative, you should go left, positive is right
+};
+
+
+var writeTrip = function(){
+  var outputString = '';
+  for (var i=0; i< trip.length - 2; i++){
+    outputString += trip[i] + ' , ';
+  }
+  if (trip.length > 1){
+    outputString += trip[trip.length-2];
+    outputString += ' and ' + trip[trip.length-1];
+  } else {
+    outputString += trip[trip.length-1];
+  }
+  return outputString;
+};
+
+
+var createTrip = function(station1, station2, line){       //this only works within the same line
+  var position1 = findPosition(line,station1);
+  var position2 = findPosition(line,station2);
+  var direction = findDirection(position1,position2);     // this is a number
+  var trip = '';
+
+  if (direction > 0 ){       //positive, so going right
+    for (var i = position1+1; i<position2+1; i++ ){
+      window.trip.push(lines[line][i]);
+    }
```

Comment:
I think you'd make less work for yourself to use `.slice()` here instead of doing it the hard way with a `for` loop.

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/slice